omod/src/main/java/org/openmrs/module/htmlwidgets/web/controller/ConceptSearchController.java

```diff
@@ -45,7 +48,20 @@ public void conceptSearch(ModelMap model, HttpServletRequest request, HttpServle
     	PrintWriter out = response.getWriter();
 
     	List<Locale> l = new Vector<Locale>();
-    	l.add(Context.getLocale());
+    	//l.add(Context.getLocale());
+    	int listIndex=0;
+    	User user = Context.getAuthenticatedUser();
+    	if(user.getUserProperties() != null){
+    		if(user.getUserProperties().containsKey(OpenmrsConstants.USER_PROPERTY_PROFICIENT_LOCALES)){
+    			List<Locale>userLocalList = new Vector<Locale>();
+    			userLocalList = new Vector (Arrays.asList(user.getUserProperty(OpenmrsConstants.USER_PROPERTY_PROFICIENT_LOCALES).split(" , ")));
+		    	while(listIndex<userLocalList.size()){
+		    	l.addAll(listIndex, userLocalList);
+		    	listIndex++;
+		    	}
+    		}
+    	}
+    	
     	List<ConceptWord> words = Context.getConceptService().getConceptWords(query, l, false, includeClasses, null, null, null, questionConcept, null, null);
```

Comment:
You are on the right track here, but this needs a little work.  I don't know the why you are iterating over the userLocaleList like this, or creating a new Vector.  Can you simply the code a bit more?  Also, you will need to find a way to test this and demonstrate that it fixes the problem before we can merge it in.  Thanks for working on this!